### PR TITLE
patch: check environment variable existence

### PR DESCRIPTION
**Pull request by raprocks**
# Description of change

Please write a summary of your changes and why you made them. 

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes #issue_number`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Hi @raprocks, thanks for the PR! Just checking in, do you have any question regarding the issue?

**Comment by thibault-martinez** [comment]
Hi @raprocks, let us know if you don't have time to finish and we'll do it

**Comment by raprocks** [comment]
hey @thibault-martinez , i got a little busy but i am back now. The PR was not for hacktoberfest haha, i will complete in this week sometime and make this ready for submission. I have the task clear. Thanks

**Comment by raprocks** [comment]
@thibault-martinez Please Review and let me know if any other changes are necessary

File: bindings/nodejs/examples/client/05-get-address-balance.ts [hunk 1]
```diff
@@ -11,10 +11,13 @@ require('dotenv').config({ path: '.env' });
...
     initLogger();
-    if (!process.env.NODE_URL) {
-        throw new Error('.env NODE_URL is undefined, see .env.example');
+    for (const envVar of ['NODE_URL', 'MNEMONIC']) {
+        if (!(envVar in process.env)) {
+            throw new Error(`.env ${envVar} is undefined, see .env.example`);
+        }
     }
 
+
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/client/05-get-address-balance.ts]:
```suggestion
```

File: bindings/nodejs/examples/client/05-get-address-balance.ts
Comment (thibault-martinez) [inline bindings/nodejs/examples/client/05-get-address-balance.ts:1]: L26 of this file, remove the check

File: bindings/nodejs/examples/client/09-transaction.ts
Comment (thibault-martinez) [inline bindings/nodejs/examples/client/09-transaction.ts:1]: L26, remove the check

File: bindings/nodejs/examples/client/14-build-foundry-output.ts
Comment (thibault-martinez) [inline bindings/nodejs/examples/client/14-build-foundry-output.ts:1]: L31 remove the check

File: bindings/nodejs/examples/client/06-simple-block.ts [hunk 2]
```diff
@@ -10,8 +10,10 @@ require('dotenv').config({ path: '.env' });
 // In this example we will send a block without a payload.
 async function run() {
     initLogger();
-    if (!process.env.NODE_URL) {
-        throw new Error('.env NODE_URL is undefined, see .env.example');
+    for (const envVar of ['NODE_URL']) {
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/client/06-simple-block.ts]:
```suggestion
    for (const envVar of ['NODE_URL', 'EXPLORER_URL']) {
```

**Comment by raprocks** [comment]
sorry missed that!

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
@raprocks there are some warnings now (you should also see them locall with `yarn lint`), because TS doesn't know the type of the values
I think if you just put `as string` in those places it should be fine again

**Review by Thoralf-M — APPROVED** [review]
Thanks!

**Review by thibault-martinez — APPROVED** [review]
Thank you :)

**Pull request merged**